mllib/src/main/scala/org/apache/spark/ml/classification/LogisticRegression.scala

```diff
@@ -407,6 +449,60 @@ private[classification] class MultiClassSummarizer extends Serializable {
   }
 }
 
+@Experimental
+class LogisticRegressionTrainingSummary private[classification] (
+    predictions: DataFrame,
+    probabilityCol: String,
+    labelCol: String,
+    val objectiveHistory: Array[Double])
+  extends LogisticRegressionSummary(predictions, probabilityCol, labelCol) {
+
+  /** Number of training iterations until termination */
+  val totalIterations = objectiveHistory.length
+
+}
+
+@Experimental
+class LogisticRegressionSummary private[classification] (
+  @transient val predictions: DataFrame,
```

Comment:
Yes.

Since models will be serialized to disk (model save/load) and sent to executors (during `model.predict` closure capture) we wanted to avoid movement of large and expensive data.

Small/cheap summary statistics (e.g. scalars precision, recall, auroc) should be available in the `*Model` instance but large ones (e.g. residuals in LinearRegressionSummary) should only be available in the driver and not be serialized (i.e. transient).